### PR TITLE
IOHKLE-204: Cardano ADA integration

### PR DESCRIPTION
**Pull request by DCRichards**
### Changes

* Integration of Cardano ADA into ledgerjs codebase and into flowtype/ECMAScript 2015, with mocha tests migrated.
* Added snapshot tests in line with other platforms.
* Added `@ledgerhq/hw-app-ada` to README.md
* .gitignore tweaks

### CODE REVIEW

File: packages/hw-app-ada/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+First, ensure you have a **test** build installed on the device (see ledger app respository for details). Then run:
+
+```shell
+yarn run core-test
+```
+
+#### API Tests
+
+These test the production API and can be run either on a production build or headlessly for fully-automated testing.
```
Comment (StaceC) [inline packages/hw-app-ada/README.md:31]: Typo

Comment (DCRichards) [reply packages/hw-app-ada/README.md:31]: *repository - nabbed it!

**Comment by StaceC** [comment]
All tests passing in test api, headless and production modes. Flipping brilliant! Two subtle ones (a) Should we link back to the C app in the readme (b) first prod test doesn't prompt the user to confirm public key generation (if that always was the case, happy to leave ftm)

**Comment by DCRichards** [comment]
@StaceC Updated according to feedback 👌

**Pull request merged**